src/Wellcome.Dds/Wellcome.Dds.Dashboard/Controllers/DashController.cs

```diff
@@ -0,0 +1,840 @@
+                model.CachedTextModelFileInfo = cacheBuster.GetAltoSearchTextCacheFileInfo(
+                    ddsId.BNumber, dgManifestation.SequenceIndex);
+                logger.Log("Finished cacheBuster.GetAltoSearchTextCacheFileInfo(..)");
+                model.CachedAltoAnnotationsFileInfo = cacheBuster.GetAllAnnotationsCacheFileInfo(
+                    ddsId.BNumber, dgManifestation.SequenceIndex);
+                logger.Log("Finished cacheBuster.GetAllAnnotationsCacheFileInfo(..)");
+                ViewBag.Log = LoggingEvent.FromTuples(logger.GetEvents());
+                if (json)
+                {
+                    return AsJson(model.GetJsonModel());
+                }
+                return View("Manifestation", model);
+            }
+            if (dgResource is IDigitisedCollection)
+            {
+                // This is the manifestation controller, not the volume or issue controller.
+                // So redirect to the first manifestation that we can find for this collection.
+                // Put this and any intermediary collections in the short term cache,
+                // so that we don't need to build them from scratch after the redirect.
+                var dgCollection = dgResource as IDigitisedCollection;
+                string redirectId;
+                PutCollectionInShortTermCache(dgCollection);
+                if (dgCollection.MetsCollection.Manifestations.HasItems())
+                {
+                    // a normal multiple manifestation, or possibly a periodical volume?
+                    if (json)
+                    {
+                        return RedirectToAction("Collection", "Dash", new { id = ddsId.BNumber, json = "json" });
+                    }
+                    redirectId = dgCollection.MetsCollection.Manifestations.First().Id;
+                    return RedirectToAction("Manifestation", "Dash", new { id = redirectId });
+                }
+                // a periodical, I think - but go to volume controller for this.
+                redirectId = dgCollection.MetsCollection.GetRootId();
+                if (json)
+                {
+                    return RedirectToAction("Collection", "Dash", new { id = redirectId, json = "json" });
+                }
+                return RedirectToAction("Collection", "Dash", new { id = redirectId });
+            }
+            ViewBag.Message = "Unknown type of resource found for identifier " + id;
+            if (json)
+            {
+                return NotFound(ViewBag.Message);
+            }
+            return View("ManifestationError");
+        }
+
+
+        private async Task<IDigitisedCollection> GetCachedCollectionAsync(string identifier)
+        {
+            var coll = await cache.GetCached(
+                CacheSeconds,
+                CacheKeyPrefix + identifier,
+                async () => (await dashboardRepository.GetDigitisedResourceAsync(identifier)));
```

Comment:
`() => dashboardRepository.GetDigitisedResourceAsync(identifier));`? Not sure how `ISimpleCache` will handle a `Task<T>` as it's not awaiting it...?

Reply:
Added some comments here for later attention